### ISSUE TITLE
Toshiba NVMe not recognized

### ISSUE THREAD

**Issue by ZombieTheBest** [syscl/XPS9350-macOS#44]
Hi! 
I can reach the installer but the Toshiba drive doesn't show up in Disk Utility. BIOS Settings are good, SATA set to AHCI. 
I've already tried your previous solution by activating the patch on the plist file, with no success. 

Thank you again!

**Comment by syscl** [comment]
@ZombieTheBest Give me more detail:

- What specs
- Which system version you use
- Do you use any other Clover and mix up with my Clover?

syscl

**Comment by ZombieTheBest** [comment]
@syscl 

1) i7-6600U
    RAM 8GB
    Intel HD520   
    Display QHD+

2) MacOS Sierra 10.12.3

3) No, just your Clover folder!

**Comment by R1MnTnA** [comment]
I believe I had the same problem as you. Please see here. - https://www.tonymacx86.com/threads/guide-dell-xps-13-9350-macos-10-12-1.204730/page-63#post-1420083

**Comment by ZombieTheBest** [comment]
@R1MnTnA I have no patch in ACPI/patched, and DropOEM is already set to false. I have tried to put all my kexts in "Other" folder too. I have tried to use the sata-100-unsupported kext too, but with no success :(

Btw, I have no "IONVMeFamily.kext" in my kext folder, is that correct?

**Comment by R1MnTnA** [comment]
You should only have whatever is in syscl's Clover folder copied to your Clover folder on the install usb drive. Did you try to make the SSDT-NVMe-Pcc.aml file?

**Comment by ZombieTheBest** [comment]
@R1MnTnA Thank you for your answer!
Unfornately I don't know where to start with the HackrNVmeFamily, maybe i will look into it in the afternoon. The file is not in syscl's Clover folder, so i don't have it. Another thing i noticed in Bios: under System Configuration->Drives i can find two tick box, one with "M2 PCI-e SSD-0" and the other one with "SATA-2". I tried to disable "SATA-2" but it still doesn't recognize the Toshiba Drive. What if it's not NVme?

**Comment by ZombieTheBest** [comment]
Okay, it's Nvme. The BIOS Device Name is \_SB.PCI0.RP09.PXSX. I'm trying to make the SSDT Nvme patch :)

**Comment by ZombieTheBest** [comment]
@R1MnTnA Ok, i followed the guide for making the SSDT-NVMe-PCC.aml file, but it still doesn't show up. Here what i've done:

1. On Windows i found the device name: _SB.PCI0.RP09.PXSX
2. I have created the SSDT-NVMe-Pcc.aml file with these lines:

`// Inject bogus class-code for NVMe SSD to prevent IONVMeFamily.kext from loading
DefinitionBlock("", "SSDT", 2, "hack", "NVMe-Pcc", 0)
{
    External(_SB.PCI0.RP09.PXSX, DeviceObj)
    Method(_SB.PCI0.RP09.PXSX._DSM, 4)
    {
        If (!Arg2) { Return (Buffer() { 0x03 } ) }
        Return(Package()
        {
            "class-code", Buffer() { 0xff, 0x08, 0x01, 0x00 },
            "built-in", Buffer() { 0 },
        })
    }
}
//EOF`

3. I have placed the file in my EFI/Clover/ACPI/patched folder
4. I have edited the config. plist file, i have removed all "Patch NVme" lines and i have added the line <string>SSDT-NVMe-Pcc.aml</string> under <array> in <key>SortedOrder</key>.
5. Using my hackintosh i generated the HackrNVMeFamily_10_12_3 file using the --spoof option. I have placed the kext file in my EFI/Clover/kext/Other folder and in my EFI/Clover/kext/10_12 folder, just to be sure. 

It still doesn't work, the ACPI folder has no other file, no DSDT. Ideas? :) 
Thank you!

**Comment by ZombieTheBest** [comment]
Okay. I just realized i had to COMPILE the aml file using maciASL, sorry... lack of experience here. Now i can finally see the NVme Disk! THANK YOU FOR YOUR HELP! 👍  🥇 
May i let this open for further help (cause i will almost surely need it)?

**Comment by R1MnTnA** [comment]
Also add 
`				<dict>
					<key>Comment</key>
					<string>Change _DSM to XDSM</string>
					<key>Find</key>
					<data>X0RTTQ==</data>
					<key>Replace</key>
					<data>WERTTQ==</data>
				</dict>
`
in the config.plist under DSDT>Patches

**Comment by R1MnTnA** [comment]
Awesome. Sure thing. 👍🏼

**Comment by ZombieTheBest** [comment]
Sure we have learned all from him :D 
And sure i will read how DSDT patching works :)

Just a question about the script: after first boot i reboot using pendrive, i press FN+F4, i get my DSDT and SSDT files in the EFI/Clover folder and i have to copy all of them somewhere inside the script folder right?
Or i have to first install Clover on my disk, and then get the files directly from the Clover installed on the internal disk? I have to place files somewhere or just leave them there? Thank you!

**Comment by ZombieTheBest** [comment]
another problem. I've installed Clover on my disk and i've copied the Pendrive Clover Directory to the EFI partition of the disk. Everything worked, so i pressed FN+F4 at boot and i got DSDT files in CLOVER/ACPI/origin. I ran the script without errors, no the system doesn't boot anymroe. In verbose mode i can't see any error, just a "forbidden" icon. I think this is related to NVMe patch, and i think that it can't find the NVme anymore. I checked that the HackrNVMeFamily_10_12_3 is still there, and the config.plist file seems okay. It still can boot from pendrive of course

**Comment by ZombieTheBest** [comment]
@R1MnTnA adding those lines in config.plist under DSDT>Patches gives a Kernel Panic: `AppleACPIPlatformExpert::start failed`, unable to find driver for this platform...

**Comment by R1MnTnA** [comment]
I believe you need that patch from the start, before install on the usb drive. Also for the HackerNVMe........kext file, did you create it using the --spoof 10_12_3 command?

**Comment by R1MnTnA** [comment]
Did you only try FN+F4 or did also try just plain F4 too. That's what I had to do to get the original dsdt & ssdt files.

**Comment by ZombieTheBest** [comment]
I got files on my Clover/ACPI/orig so i guess it worked, right? :)
Yes, the HackerNVMe was created using --spoof command. 
I might need to apply the patch and remove _DSM from the DSDT file and all SSDT's right?  I'm trying to do that, i renamed all the lines in SSDT files and recompiled. DSDT doesn't compile (even before i apply the patch), i got three errors. I already tried to remove all the lines causing compile error, but when i do that i got more errors after I use the patch.

**Comment by R1MnTnA** [comment]
Ok, I thought possibly you had taken what syscl had in his origin folder. Sounds like the FN+F4 is working. I think there is a setting in the bios that allows you not to have hit FN but just The F keys. Anyway, do not manually go into the files and change them. Just add that piece of code above into the patches section of config. Also I noticed syscl had told you to change the IONVME code from true to false in the config. I didn't have to do that, so make sure that is set back to true, but keep the DropOEM to false.

**Comment by R1MnTnA** [comment]
Also are you starting with a clean formated NVMe ssd? Not that that will make any difference of it showing up, but more for when you go to install it.

**Comment by ZombieTheBest** [comment]
@R1MnTnA 
DropOEM is set to FALSE. I don't have anymore the IONVME code in my PLIST file (i used the HackedIONVME and rebaham said to remove everything from plist file so I did!).

**Comment by ZombieTheBest** [comment]
@R1MnTnA If I add the piece of code above into the Patches section i get Kernel Panic at boot

**Comment by syscl** [comment]
You don't need _DSM -> XDSM which will cause panic after exciting deploy.sh.

deploy.sh has already applied this patch and will smartly inject required _DSM for XPS 13.

syscl

**Comment by ZombieTheBest** [comment]
@syscl FIXED IT! I placed your original config.plist, removed HackrIONVme kext and it WORKED! 
WOW THANK YOU!!!!🥇

**Comment by ZombieTheBest** [comment]
Okay nice! I confirm that everything is working 👍 
Last thing: Intel Wifi Card doesn't work, I guess I have to replace it, which one I should buy? :)

**Comment by syscl** [comment]
@ZombieTheBest DW 1830 or DW 1560

syscl

**Comment by ZombieTheBest** [comment]
just one more thing: i have some glitches with touchpad, and the "Trackpad" option in settings doesn't show anything

**Comment by syscl** [comment]
@ZombieTheBest Choose VoodooPS2 instead ApplePS2SmartTouchPad in deploy, then reboot.

syscl

**Comment by ZombieTheBest** [comment]
@syscl After reboot i lost Brightness keys and still no trackpad. I reinstall everything from zero now that i know how to get it working! Thank you! :D

**Comment by syscl** [comment]
deploy will remove ApplePS2SmartTouchPad and inject keys information for VoodooPS2 if you choose VoodooPS2 in deploy. That's why I said you need to choose kext in deploy means.

syscl

**Comment by ZombieTheBest** [comment]
@syscl @R1MnTnA THANK YOU BOTH! 💯 
Everything (except WiFi) is now working! :D 

I will buy the internal card you told me, meanwhile i'm trying to get an 801.11 USB WiFi adapter working. Thank you again! 👍

**Comment by syscl** [comment]
@ZombieTheBest Nice to hear that, if you can share your method that will be much more appreciated :)

syscl

**Comment by ZombieTheBest** [comment]
Sure! This is what i have done:
1. I have created the USB Bootable with Sierra 10.11.3, but I couldn't see my Toshiba NVme drive on Disk Utility

2. Solved with HackrNVMeFamily.kext, made by following this guide:
https://www.tonymacx86.com/threads/guide-hackrnvmefamily-co-existence-with-ionvmefamily-using-class-code-spoof.210316/

3. I finally got into Mac, installed Clover. I thought I needed the HackrNVMeFamily.kext to boot the Mac without pendrive, but I was wrong: the only thing I had to do was to use your CLOVER folder.

4. Reboot, pressed FN+F4 on Clover to dump DSDT & SSDT's. 

5. Launched the script selecting "2. Voodoo PS2", reboot again.

And everything is now working! :D 
I managed the USB Wifi to work too until i get the internal Wireless card. Unfortunately wifi dongle is not recognized after sleep and needs reboot, it's a Ralink RT5730, I applied the text (with modified VID and PID), and it's working with DWA-140 pref panel. Any ideas to make it working after sleep? :)
